src/stream/src/from_proto/source.rs

```diff
@@ -57,17 +60,23 @@ impl ExecutorBuilder for SourceExecutorBuilder {
             Field::with_name(column_desc.data_type.clone(), column_desc.name.clone())
         }));
         let schema = Schema::new(fields);
-        let keyspace = Keyspace::table_root(store, &TableId::new(node.state_table_id));
+
         let vnodes = params
             .vnode_bitmap
             .expect("vnodes not set for source executor");
 
+        let state_table_handler = SourceStateTableHandler::from_table_catalog(
+            node.state_table.as_ref().unwrap(),
+            store,
+            Some(Arc::new(vnodes.clone())),
+        );
```

Comment:
IIRC, our `SourceExecutor` does not satisfy consistent hash. so here the vnode should be None :thinking: I believe even the `SourceStateTableHandler::from_table_catalog()` should not have the params.